sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.ControlFlowGraph;
+using SonarAnalyzer.ControlFlowGraph.CSharp;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.ShimLayer.CSharp;
+
+namespace SonarAnalyzer
+{
+    public static class SyntaxNodeExtension
```

Comment:
if this is public, it should be in its own file
why not put this together with the other extension methods?

Reply:
Is switching it to `internal` enough to you?
I'd like to avoid making it accessible to anything but `MLIRExporter`, because what it does looks very specific to our use case to me...